File: src/cmcandy/C_Language_Answers/_0096.cpp
#include <iostream>
#include <vector>
using namespace ::std;
class Solution
{
public:
    int numTrees(int n)
    {
        long long c = 1;
        for (int i = 0; i < n; i++)
        {
            c = c * 2 * (2 * i + 1) / (i + 2);
        }
        return (int)c;
    }
};

File: src/cmcandy/Python_language_Answers/_0096.py
class Solution:
    def catalan(self, n: int) -> int:
        if n == 0 or n == 1:
            return 1
        return (4 * n - 2) * self.catalan(n - 1) // (n + 1)

    def numTrees(self, n: int) -> int:
        return self.catalan(n)